packages/story-editor/src/app/media/utils/getResourceFromLocalFile.js

```diff
@@ -84,19 +84,9 @@ const getVideoResource = async (file) => {
   const canPlayVideo = '' !== videoEl.canPlayType(mimeType);
   if (canPlayVideo) {
     videoEl.src = src;
-    videoEl.addEventListener('loadedmetadata', () => {
-      length = Math.round(videoEl.duration);
-      const seconds = formatDuration(length % 60);
-      let minutes = Math.floor(length / 60);
-      const hours = Math.floor(minutes / 60);
-
-      if (hours) {
-        minutes = formatDuration(minutes % 60);
-        lengthFormatted = `${hours}:${minutes}:${seconds}`;
-      } else {
-        lengthFormatted = `${minutes}:${seconds}`;
-      }
-    });
+    const videoLength = await getVideoLength(src);
+    length = videoLength.length;
+    lengthFormatted = videoLength.lengthFormatted;
```

Comment:
```suggestion
    const { length, lengthFormatted } = await getVideoLength(src);
```

Along with the catch statement I suggest, this should make this cleaner.